samples/bluetooth/hci_usb/README.rst

```diff
@@ -21,3 +21,28 @@ This sample can be found under :zephyr_file:`samples/bluetooth/hci_usb` in the
 Zephyr tree.
 
 See :ref:`bluetooth samples section <bluetooth-samples>` for details.
+
+Notes on HCI over USB
+*********************
+
+USB is a serial protocol. However, the implementation of the peripheral can
+expose multiple endpoints, in effect re-ordering serial data into per-endpoint
+receive queues.
+
+The Bluetooth specification defines separate endpoints per HCI packet type.
+
+As the HCI protocol was not originally designed with re-ordering in mind, using
+this transport may expose the application to deadlocks and general unexpected
+behavior.
+
+Because of this, the industry-standard way to expose a Bluetooth Controller over
+USB is now to use the H4 HCI transport over virtual serial (i.e. CDC class).
+
+See the :zephyr:code-sample:`HCI UART sample <bluetooth-hci-uart-sample>`
+documentation for more details.
+
+If using the controller with a BlueZ host, one can also use the USB HCI H4
+transport. It exposes the H4 HCI transport, but over two custom bulk endpoints.
```

Comment:
Is it really the case? Was USB H4 added to btusb driver? Can you please add a link and commit SHA?

Reply:
@jhedberg do you know when that was added?

Reply:
Yes, this is correct @jfischer-no. Only BlueZ impements "USB H4", other USB Hosts do not.